ros2action/ros2action/verb/send_goal.py

```diff
@@ -121,6 +123,32 @@ def send_goal(action_name, action_type, goal_values, feedback_callback):
 
         goal_handle = goal_future.result()
 
+        # install signal handler to cancel the goal on SIGINT
+        def _sigint_cancel_handler(sig, frame):
+            nonlocal goal_handle
```

Comment:
Thank you for working on this so promprtly, and I'll test when I get a chance.

For now, I wanted to say you can avoid the `nonlocal` with `partial` if you'd prefer:

https://stackoverflow.com/a/65277685

Reply:
i would keep current if no clear benefits there, and there are some places use nonlocal too. thanks for your effort.